akka-remote/src/test/scala/akka/remote/artery/DeathWatchNotificationSpec.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.artery
+
+import scala.concurrent.duration._
+
+import akka.actor._
+import akka.testkit._
+import com.typesafe.config.ConfigFactory
+
+object DeathWatchNotificationSpec {
+
+  val config = ConfigFactory.parseString(s"""
+    akka {
+        loglevel = DEBUG
+        actor {
+            provider = remote
+        }
+        remote.use-unsafe-remote-features-outside-cluster = on
+    }
+    """).withFallback(ArterySpecSupport.defaultConfig)
+
+  object Sender {
+    def props(receiver: ActorRef, sendOnStop: Vector[String]): Props =
+      Props(new Sender(receiver, sendOnStop))
+  }
+
+  class Sender(receiver: ActorRef, sendOnStop: Vector[String]) extends Actor {
+    override def receive: Receive = {
+      case msg => sender() ! msg
+    }
+
+    override def postStop(): Unit = {
+      sendOnStop.foreach(receiver ! _)
+    }
+  }
+}
+
+class DeathWatchNotificationSpec extends ArteryMultiNodeSpec(DeathWatchNotificationSpec.config) with ImplicitSender {
+  import DeathWatchNotificationSpec.Sender
+
+  private val otherSystem = newRemoteSystem(name = Some("other"))
+
+  private val messages = (1 to 100).map(_.toString).toVector
+
+  private def setupSender(receiverProbe: TestProbe, name: String): Unit = {
+    val receiverPath = receiverProbe.ref.path.toStringWithAddress(address(system))
+    val otherProbe = TestProbe()(otherSystem)
+    otherSystem.actorSelection(receiverPath).tell(Identify(None), otherProbe.ref)
+    val receiver = otherProbe.expectMsgType[ActorIdentity](5.seconds).ref.get
+    receiver.path.address.hasGlobalScope should ===(true) // should be remote
+    otherSystem.actorOf(Sender.props(receiver, messages), name)
+  }
+
+  private def identifySender(name: String): ActorRef = {
+    system.actorSelection(rootActorPath(otherSystem) / "user" / name) ! Identify(None)
+    val sender = expectMsgType[ActorIdentity](5.seconds).ref.get
+    sender
+  }
+
+  "receive Terminated after ordinary messages" in {
+    val receiverProbe = TestProbe()
+    setupSender(receiverProbe, "sender")
+    val sender = identifySender("sender")
+
+    receiverProbe.watch(sender)
+    // make it likely that the watch has been established
+    sender.tell("echo", receiverProbe.ref)
+    receiverProbe.expectMsg("echo")
+
+    sender ! PoisonPill
+    receiverProbe.receiveN(messages.size).toVector shouldBe messages
```

Comment:
This test fails every time without the flushing. Often the Terminated arrives as the second message.